Prod openstack (infra)

joshmoore opened the pull request:
Few cleanup commits as well as a few new playbooks for getting new production services on openstack:

 * `training-server` migrated from management_tools (thanks @kennethgillen!)
 * playbooks for production docker containers
 * playbooks for development web server

manics commented on ansible/os-uod-docker.yml:
```diff
@@ -0,0 +1,95 @@
+
+  tasks:
+
+    - set_fact: vm_joined_groups="{{ vm_groups.split(',') | union(vm_extra_groups.split(',')) | join(',')}}"
```
Is this useful?

manics commented on ansible/os-uod-docker.yml:
```diff
@@ -0,0 +1,95 @@
+        ansible_host: "{{ vmdocker.server.public_v4 }}"
+        ansible_user: "centos"
+        ansible_become: true
+      when: vmdocker.changed
```
Is this copied from the IDR playbooks? This is only needed if you want to create VMs and deploy servers in the same playbook, and is one of the things we discussed regarding unnecessary complexity with the IDR deployments. Just wanted to double-check you're happy with that here (it's self contained, so not such a big problem).

manics commented on ansible/training-server.yml:
```diff
@@ -0,0 +1,84 @@
+  tasks:
+  - name: training-server | add operating system group for trainers
+    become: true
+    group: name=trainers state=present
```
How pedantic are you feeling? :-) (short/long form)

manics commented on ansible/training-server.yml:
```diff
@@ -0,0 +1,84 @@
+
+  - name: training-server | add operating system group for facility managers
+    become: true
+    group: name=facility-managers state=present
```
How pedantic are you feeling? :-) (short/long form)

manics commented on ansible/uod-docker.yml:
```diff
@@ -0,0 +1,18 @@
+  - role: docker
+    docker_groupmembers: [centos]
+  - role: sudoers
+  - role: local-accounts
```
Are there some variables (possibly private?) to go with the `sudoers` and `local-accounts` roles? They don't do anything on their own.

joshmoore replied:
Not yet. Vagrant is still not passing. Currently DB creation appears to be missing. Once I'm passed that, I'll look into testing variables.

kennethgillen replied:
@joshmoore - giving the vagrant VM more ram is enough to let the playbook get past the systemd startup stage - this failed for me the first time I tried vagrant provisioning, too.

```
LS27172:ansible kenny$ git diff -p
diff --git a/ansible/Vagrantfile b/ansible/Vagrantfile
index 73e6cf2..1b4a5db 100644
--- a/ansible/Vagrantfile
+++ b/ansible/Vagrantfile
@@ -5,6 +5,12 @@
 # See: http://docs.ansible.com/ansible/guide_vagrant.html
 Vagrant.require_version ">= 1.7.0"

+Vagrant.configure("2") do |config|
+  config.vm.provider "virtualbox" do |vb|
+    vb.customize ["modifyvm", :id, "--memory", "1024"]
+  end
+end
+
 Vagrant.configure(2) do |config|

   config.vm.define "docker" do |docker|
```

joshmoore replied:
I'll try this now. The failure I've been getting is:

```
TASK [omero-server : omero | enable server systemd] ****************************
fatal: [training-server]: FAILED! => {"changed": false, "failed": true, "msg": "Job for omero.service failed. See 'systemctl status omero.service' and 'journalctl -xn' for details.\n"}
```

```
~/opt/management_tools/infrastructure/ansible $ vagrant ssh training-server
Last login: Thu Jul 14 23:44:12 2016 from 10.0.2.2
[vagrant@localhost ~]$ sudo systemctl status omero.service
omero.service - OMERO.server
   Loaded: loaded (/etc/systemd/system/omero.service; enabled)
   Active: failed (Result: timeout) since Thu 2016-07-14 23:49:12 EDT; 25s ago
  Process: 19628 ExecStart=/home/omero/OMERO.server/bin/omero admin start (code=killed, signal=TERM)
```

```
Caused by: org.postgresql.util.PSQLException: FATAL: password authentication failed for user "omero"
```

manics replied:
can you try and manually login to postgres from that shell?

manics replied:
also `systemctl status XXX.service -l` may give more lines of context

joshmoore replied:
Fresh run with 2g of RAM:

```
TASK [omero-server : omero | install omero] ************************************
fatal: [training-server]: FAILED! => {"changed": true, "cmd": ["/home/omero/omego/bin/omego", "install", "--release", "latest", "--sym", "OMERO.server", "--prestartfile", "/home/omero/config/omero-base.config", "--prestartfile", "/home/omero/config/omero-additional.config", "-qq", "--ice", "3.5", "--no-start", "--dbhost", "localhost", "--dbuser", "omero", "--dbname", "omero", "--dbpass", "omero", "--rootpass", "omero", "--initdb"], "delta": "0:02:54.722766", "end": "2016-07-15 08:55:04.728419", "failed": true, "rc": 30, "start": "2016-07-15 08:52:10.005653", "stderr": "2016-07-15 08:55:04,664 [omego.extern] ERROR Failed [0.009 s]\n2016-07-15 08:55:04,664 [    omego.db] ERROR Non-zero return code\ncommand: psql -d omero -h localhost -U omero -w -A -t -c \\conninfo\nreturn code: 2\nstdout: \nstderr: psql: FATAL:  password authentication failed for user \"omero\"", "stdout": "ERROR: Database connection check failed", "stdout_lines": ["ERROR: Database connection check failed"], "warnings": []}
```

kennethgillen replied:
Could my ansible checkout be pulling in some vars from somewhere, @manics? How can I test this in isolation? It really seems to work for me..

joshmoore replied:
> can you try and manually login to postgres from that shell?

```
~/opt/management_tools/infrastructure/ansible $ vagrant ssh training-server
Last login: Fri Jul 15 09:49:46 2016 from 10.0.2.2
[vagrant@localhost ~]$ sudo -u omero -s
[omero@localhost vagrant]$ psql -l
psql: FATAL:  role "omero" does not exist
```

manics replied:
`sudo -iu postgres psql -l `?

joshmoore replied:
```
[vagrant@localhost ~]$ sudo -iu postgres psql -l
                                  List of databases
   Name    |  Owner   | Encoding |   Collate   |    Ctype    |   Access privileges
-----------+----------+----------+-------------+-------------+-----------------------
 postgres  | postgres | UTF8     | en_US.UTF-8 | en_US.UTF-8 |
 template0 | postgres | UTF8     | en_US.UTF-8 | en_US.UTF-8 | =c/postgres          +
           |          |          |             |             | postgres=CTc/postgres
 template1 | postgres | UTF8     | en_US.UTF-8 | en_US.UTF-8 | =c/postgres          +
           |          |          |             |             | postgres=CTc/postgres
(3 rows)
```

joshmoore replied:
Discussed in sysadmin: turns out to be a parsing issue in ansible!

manics commented:
Can the variables in 26b4aa1 be overridden by a group/host-vars file?

joshmoore commented:
Yup:

```
[vagrant@localhost sync]$ ansible-playbook -i host training-server.yml -C -e @vars.yml
...
TASK [training-server | add facility manager operating system users] ***********
changed: [localhost] => (item=bob1)

TASK [training-server | add symlinks to sample images in fm users homedirs] ****
changed: [localhost] => (item=bob1)

TASK [training-server | add OMERO.DropBox folders] *****************************
changed: [localhost] => (item=bob1)

TASK [training-server | allow OMERO user sudo by list of users in variable] ****
ok: [localhost] => (item=centos)

TASK [training-server | allow OMERO service control by list of users in variable] ***
ok: [localhost] => (item=centos)

PLAY RECAP *********************************************************************
localhost                  : ok=66   changed=7    unreachable=0    failed=0

[vagrant@localhost sync]$ cat vars.yml
users_to_be_added:
    - bob1
```

manics commented:
Couple of notes:
- need to check whether inventory host/group vars will override the playbook vars
- long vs short form yaml styles in ansible/uod-web.yml

joshmoore commented:
Agreed about the cleanup, but merging this as the current state of the `ci` tenancy. Passing off to others to make optimal use of that. (I'll bump the submodule before merging 167)

joshmoore commented:
Local changes which were being tested to get all of this working:

```
/opt/management_tools/infrastructure $ git diff
diff --git a/ansible/uod-web.yml b/ansible/uod-web.yml
index e357977..a6ca0f5 100644
--- a/ansible/uod-web.yml
+++ b/ansible/uod-web.yml
@@ -14,17 +14,23 @@
     live_site_aliases:
       - openmicroscopy.org
   roles:
-    - nginx
+    - role: versioncontrol-utils
+    - role: nginx
+    - role: jenkinsslave
   pre_tasks:
-    - name: install rubygems for jekyll
-      yum: name=rubygems state=present
-      become: yes
-    - name: install ruby-devel
-      yum: name=ruby-devel state=present
+    - name: install required packages
       become: yes
+      yum:
+        name: "{{ item }}"
+        state: present
+      with_items:
+       - rubygems
+       - ruby-devel
+       - make
+       - gcc
   tasks:
     - name: Install jekyll
-      gem: name=jekyll state=present user_install=no
+      gem: name=jekyll state=present user_install=no include_doc=no
       become: yes

     - name: dev website deploy directory
```

cc: @sbesson
